modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -206,6 +213,32 @@ private static long getRequestCount(Request<?> request) {
         return requestCount.longValue();
     }
 
+    /**
+     * Used for APM style metrics to measure statics about performance. This is not for billing.
+     */
+    private static long getHttpRequestTimeInMicros(Request<?> request) {
+        List<TimingInfo> requestTimesIncludingRetries;
+        requestTimesIncludingRetries = request.getAWSRequestMetrics()
+            .getTimingInfo()
+            .getAllSubMeasurements(AWSRequestMetrics.Field.HttpRequestTime.name());
```

Comment:
`HttpRequestTime` including time spent on acquiring an existing or creating a new connection, marshalling the input etc. Do we want to include for measuring "time to first byte"?

IIUC, last time we discussed about it, we prefer to exclude time spent on local node and measure only for networking, e.g. something like the sum of `HttpClientSendRequestTime` and `HttpClientReceiveResponseTime`. Maybe the preference has since changed?

Reply:
Thanks for the well-considered comment @ywangd !

@original-brownbear I think this one's for you.  We discussed using `TTFB` (in addition to another metric that measures the total time to fetch the block) as a way to infer what portion of the latency is  on us vs. what's on AWS.  I do see @ywangd 's point regarding the measurement of the total networking time.  

I suppose that these submeasurements compose an overlapping Venn diagram containing `Our latency` + ( `Our latency` + `AWS latency`) + `AWS latency`. If we want to know why things are slow (us vs AWS), the tricky bit is getting signal given the fact that there's overlap in responsibility for a number of these  (`Our latency` + `AWS latency`).  

Seeing as you've got a better background+understanding of where the overlap exists, WDYT?


From the docs:
 [HttpRequestTime](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/util/AWSRequestMetrics.Field.html#HttpRequestTime)
> Number of milliseconds taken to acquire a connection from the connection pool (or create a new connection), send the request (including reading the user's input stream for streaming uploads), and receive the response status and headers. This does not include the time needed to download the response payload from the AWS service.

[HttpClientSendRequestTime](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/util/AWSRequestMetrics.Field.html#HttpClientSendRequestTime)
> Number of milliseconds taken to write the request headers and payload to the underlying socket that is connected to AWS. This does not include any time spent acquiring a connection from the connection pool (or creating a new connection).